android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt

```diff
@@ -90,9 +92,17 @@ class RNIapModule(
             return
         }
 
-        billingClient = builder.setListener(this).build()
+        if (billingClientCache?.isReady == true) {
+            Log.i(
+                TAG,
+                "Already initialized, you should only call initConnection() once when your app starts"
+            )
+            promise.safeResolve(true)
+            return
+        }
+        billingClientCache = builder.setListener(this).build()
 
-        billingClient.startConnection(
+        billingClientCache?.startConnection(
```

Comment:
Do `billingClientCache` here to be optional?

Reply:
The reason is that smartcast doesn't know that the nullable instance is no longer null. However, I found a very nice way to resolve it by using the `also` operator. It looks cleaner now. please take a look